### ISSUE TITLE
Refactor ComparisonStrategy to have a single responsiblity

### ISSUE THREAD

**Issue by alexruiz** [alexruiz/fest-assert-2.x#111]
ComparisonStrategy's responsibility is to compare 2 objects, nothing more. Right now this interface has methods that should be in utility classes related to Iterables, arrays and strings.

**Comment by joel-costigliola** [comment]
Yeah I know ... but if you move string comparison to `Strings` then `Strings` will depend on `ComparisonStrategy` (same thing with `Iterables` and `Arrays`).  At the time I felt it wasn't right but if you consider it to be better, I won't disagree.
